lark/common.py

```diff
@@ -1,15 +1,28 @@
 from copy import deepcopy
+import sys
 from types import ModuleType
 from typing import Callable, Collection, Dict, Optional, TYPE_CHECKING
 
 if TYPE_CHECKING:
     from .lark import PostLex
+    from .lexer import Lexer
+    from typing import Union, Type
+    if sys.version_info >= (3, 8):
+        from typing import Literal
+    else:
+        from typing_extensions import Literal
+    if sys.version_info >= (3, 10):
+        from typing import TypeAlias
+    else:
+        from typing_extensions import TypeAlias
 
 from .utils import Serialize
 from .lexer import TerminalDef, Token
 
 ###{standalone
 
+_ParserArgType: 'TypeAlias' = 'Literal["earley", "lalr", "cyk", "auto"]'
+_LexerArgType: 'TypeAlias' = 'Union[Literal["auto", "basic", "contextual", "dynamic", "dynamic_complete"], Type[Lexer]]'
```

Comment:
Extract type definition to a place where they can be used in multiple places. Since we aren't making `typing_extensions` a runtime dependencies, the definitions need to be a string to defer the evaluation. Which then requires the definitions to be explicitly marked as `TypeAlias`s, requiring an additional `typing_extensions` usage during type checking.